### PR TITLE
⚡️ perf: Bundle optimization!

### PR DESCRIPTION
**Pull request by insulineru**
Check the diff on pictures

**`main` branch**
image

**This branch** 
image

**Changelog:**
- Change bundler from esbuild to rollup
- Configure `sideEffects`, use pure functions everywhere, when it's possible
- Remove dependencies from ESM bundle (they can be bundled in frontend framework)
- Change `ENUMS` to `Constants`. [Enum from Typescript does not support tree-shaking](https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -26,6 +29,10 @@
     "@typescript-eslint/consistent-type-definitions": [
       "error",
       "type"
-    ]
+    ],
+    /**
+     * ESLint import правила
```
Comment (insulineru) [inline .eslintrc]:
```suggestion
     * ESLint import rules
```

File: src/context/WalletProvider.tsx [hunk 2]
```diff
@@ -744,13 +749,14 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
     // todo: sendTx reject => false
     console.log('[Wallet] sendTx', transaction)
 
-    const isSolanaTransaction = transaction instanceof Transaction
+    const isSolanaTransaction = 'partialSign' in transaction
```
Comment (nrjshka) [inline src/context/WalletProvider.tsx:755]: Why?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@viaprotocol/web3-wallets",
-  "version": "1.0.30",
+  "type": "module",
```
Comment (webistomin) [inline package.json]: Maybe I'm wrong, but it seems that this package can be used in another package only with the `"type": "module"`

I don't think it's necessary

File: package.json [hunk 4]
```diff
@@ -59,11 +76,14 @@
...
   ],
   "peerDependencies": {
+    "ethers": "^5.6.8",
     "react": "^18.1.0"
   },
   "dependencies": {
     "@coinbase/wallet-sdk": "^3.3.0",
+    "@cosmjs/launchpad": "0.24.1",
     "@cosmjs/stargate": "^0.27.0",
+    "@ethersproject/providers": "5.6.8",
```
Comment (webistomin) [inline package.json]: I think it's worth adding caret to the version

File: package.json [hunk 5]
```diff
@@ -75,19 +95,21 @@
...
     "@walletconnect/web3-provider": "^1.7.8",
-    "ethers": "^5.6.8",
     "ismobilejs": "^1.1.1",
     "long": "^5.2.0",
-    "path": "^0.12.7",
     "rango-sdk": "^0.0.5"
   },
   "devDependencies": {
     "@antfu/eslint-config": "^0.25.1",
+    "@babel/cli": "^7.18.10",
```
Comment (webistomin) [inline package.json]: package is not used

Comment (insulineru) [reply package.json]: Really 😮 
Thanks for the correction :)

File: tsconfig.json [hunk 6]
```diff
@@ -1,34 +1,33 @@
...
     "esModuleInterop": true,
+    "incremental": true,
     "isolatedModules": true,
-    "noImplicitAny": true,
+    "jsx": "react-jsx",
+    "lib": ["es2019", "es2017", "dom"],
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "noEmit": true,
+    "noImplicitAny": false,
```
Comment (iower) [inline tsconfig.json]: Is there _any_ reason to reduce type strictness?

File: package.json [hunk 7]
```diff
@@ -75,19 +95,21 @@
...
     "@types/node-fetch": "^2.6.2",
     "@types/react": "^18.0.12",
-    "esbuild": "^0.14.42",
-    "esbuild-node-externals": "^1.4.1",
+    "agadoo": "^2.0.0",
+    "babel-plugin-annotate-pure-calls": "^0.4.0",
```
Comment (iower) [inline package.json]:
```suggestion
```
This dependency also looks unused

**Review by iower — APPROVED** [review]

**Review by nrjshka — APPROVED** [review]

File: src/context/WalletProvider.tsx (hunk 2 above)
Comment (insulineru) [reply src/context/WalletProvider.tsx:755]: > Why?

`instanceof` requires the import of a module, but my version checks for the presence of a property, but requires only the type to import

**Pull request merged**